source/core/TypeAndDataManager.cpp

```diff
@@ -778,14 +786,22 @@ Boolean TypeCommon::IsA(TypeRef otherType, Boolean compatibleStructure)
                 otherEncoding == kEncoding_Ascii || otherEncoding == kEncoding_Unicode) {
                 bMatch = TopAQSize() == otherType->TopAQSize();
             }
-        } else if (thisEncoding == kEncoding_Cluster && otherEncoding == kEncoding_Cluster &&
-                   !this->IsIntrinsicClusterDataType() && !otherType->IsIntrinsicClusterDataType()) {
-            if (this->SubElementCount() == otherType->SubElementCount()) {
-                for (Int32 i = 0; i < this->SubElementCount(); i++) {
-                    if (!this->GetSubElement(i)->CompareType(otherType->GetSubElement(i)))
-                        break;
+        } else if (thisEncoding == kEncoding_Cluster && otherEncoding == kEncoding_Cluster) {
```

Comment:
I see that this is growing quite a bit now that "TODO(PaulAustin): Consider merging this function with IsA function below." should be addressed sooner. I would like to do it in a separate PR and validate it independently as a release as it seems to require both refactoring and addressing functional differences and do not want to complicate this changeset that fixes a couple of isolated issues.